### PR TITLE
createClassResolver: support resolving partial classes at runtime

### PR DESCRIPTION
**Pull request by xugao**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes
see use case in #13435

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/react-compose/src/createClassResolver.ts [hunk 1]
```diff
@@ -14,43 +23,74 @@ import { GenericDictionary, ClassDictionary } from './types';
...
-          state[enumName] === enumValue && modifiers.push(classValue);
-        } else {
-          state[key] && modifiers.push(classValue);
-        }
+export const createClassResolver = (staticClasses: ClassDictionary) => {
+  const cacheMap: RecursiveMemoNode = new Map();
+  let cacheMissCount = 0;
+  let disableCaching = false;
+
+  return (
```
Comment (dzearing) [inline packages/react-compose/src/createClassResolver.ts:31]: So, with static classes, if we know ahead of time the slots, we can actually pull out the deps automatically without them being provided. Basically every key in the class map that isn't a slot would be the key to depend on in `state`

Comment (dzearing) [reply packages/react-compose/src/createClassResolver.ts:31]: we'd just need to know slots at creation time.

Comment (khmakoto) [reply packages/react-compose/src/createClassResolver.ts:31]: @dzearing I'm guessing this is what you're doing in #13448.

Comment (xugao) [reply packages/react-compose/src/createClassResolver.ts]: I removed the memo changes from this PR for now

**Comment by xugao** [comment]
thanks for reviewing @khmakoto ! David suggested a much better alternative: https://github.com/microsoft/fluentui/pull/13493
So im going to close this PR

**Pull request closed without merging**